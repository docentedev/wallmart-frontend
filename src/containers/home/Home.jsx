import React from 'react'
import styles from './Home.module.css'
import Product from '../../components/product/Product'
import useFetchProducts from '../../hooks/useFetchProducts'

function Home() {
    const { items: products } = useFetchProducts()

    return (
        <div>
            <section className={styles.area_products}>
                {products.length > 0 ? products.map((e, i) => (
                    <Product key={i} product={e} />
                )) : (
                        <div>Cargando...</div>
                    )}
            </section>
        </div>
    );
}

export default Home;
