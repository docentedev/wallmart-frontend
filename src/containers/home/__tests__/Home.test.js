import React from 'react'
import TestRenderer from 'react-test-renderer'

import Home from '../Home'

jest.mock('../../../components/product/Product', () => ({
    __esModule: true,
    default: () => (<div className="product">Product</div>),
}))

jest.mock('../../../hooks/useFetchProducts', () => ({
    __esModule: true,
    default: () => ({ items: [{}] }),
}))

describe('Home Test', () => {
    it('should render Home without error', () => {
        const testRenderer = TestRenderer.create(<Home />)
        const testInstance = testRenderer.root

        expect(testRenderer.toJSON()).toMatchSnapshot()
        expect(testInstance.findByProps({ className: 'product' }).children).toEqual(['Product'])
    })
})
