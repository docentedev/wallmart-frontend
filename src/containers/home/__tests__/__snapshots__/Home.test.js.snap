// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test should render Home without error 1`] = `
<div>
  <section
    className="area_products"
  >
    <div
      className="product"
    >
      Product
    </div>
  </section>
</div>
`;
